numpy/testing/tests/test_utils.py

```diff
@@ -1510,13 +1510,17 @@ class ReferenceCycleInDel(object):
             cycles whenever it's garbage-collected and its __del__ runs
             """
             make_cycle = True
+            count = 0
 
             def __init__(self):
                 self.cycle = self
 
             def __del__(self):
                 # break the current cycle so that `self` can be freed
                 self.cycle = None
+                ReferenceCycleInDel.count += 1
+                if ReferenceCycleInDel.count > 5:
+                    ReferenceCycleInDel.make_cycle = False
```

Comment:
I do not think this works. It is supposed to hit a 100 depth limit and then run into the `RuntimeError`, but as is, I think it runs into the AssertionError instead. Maybe that AssertionError path should make sure it is on py2 to make the test more meaningful.

We could reduce the 100 iteration limit maybe (it does seem rather high). But probably the test is just slow, and we can mark it as such?

Reply:
marked as slow

Reply:
Sounds good, could adapt the testing code to fail on Py3, but it is a different issue.